### PR TITLE
Petitions

### PR DESCRIPTION
**Pull request by tzhenia**

### CODE REVIEW

**Comment by TrofimchukYarik** [comment]
Merging with master is not allowed

File: src/main/resources/static/src/views/Petitions/PetitionAdd.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            description : this.state.petition.description,
+            initialDate : this.state.petition.initialDate,
+            category : this.state.petition.category
+        };
+
+        const _this = this;
+        axios.post('/api/petitions/add', petition)
+            .then(function () {
+                _this.setState({registrationIsSuccessful: true});
+                _this.setState({registrationIsNotSuccessful: false});
```
Comment (gefasim) [inline src/main/resources/static/src/views/Petitions/PetitionAdd.js]: PettitionAdd does not have a _registrationSuccessful_ state.

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PetitionController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package ua.softserve.rv_028.issuecitymonitor.controller;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ua.softserve.rv_028.issuecitymonitor.dto.PetitionDto;
+import ua.softserve.rv_028.issuecitymonitor.service.PetitionService;
+
+
+// ???
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PetitionController.java]: Remove redundant comment

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/dto/PetitionDto.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    private long id;
+    private UserDto userDto;
+    private String title;
+    private String description;
+    private String initialDate;
+    private PetitionCategory category;
+
+    public PetitionDto() {}
+
+    public PetitionDto(Petition entity) {
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/dto/PetitionDto.java]: Dto doesn't know anything about Entity. You should use the _MapperService_ for transfer your entity to dto.

File: src/main/resources/static/src/views/Petitions/PetitionAdd.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                swal({title: "Petition record saved", icon: "success"})
+            })
+            .catch(function () {
+                _this.setState({registrationIsSuccessful: false});
```
Comment (TrofimchukYarik) [inline src/main/resources/static/src/views/Petitions/PetitionAdd.js]: Unused booleans

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/service/PetitionService.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+
+public interface PetitionService {
+    void deleteById(long id);
+    List<PetitionDto> findAll();
+    PetitionDto findById(long id);
+    PetitionDto update(PetitionDto petitionDto);
+
+    void regPetition(PetitionDto petitionDto);
```
Comment (TrofimchukYarik) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/service/PetitionService.java]: This should be named as a CRUD operation. (create/add)

**Review by TrofimchukYarik — APPROVED** [review]

**Review by Michaluch — CHANGES_REQUESTED** [review]
Could fix all the your code snippets, which is commented for you, please.

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PetitionController.java (hunk 2 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PetitionController.java]: Done, thx!

File: src/main/resources/static/src/views/Petitions/PetitionAdd.js (hunk 1 above)
Comment (tzhenia) [reply src/main/resources/static/src/views/Petitions/PetitionAdd.js]: Done, thx!

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/service/PetitionService.java (hunk 5 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/service/PetitionService.java]: ok, Done!

File: src/main/resources/static/src/views/Petitions/PetitionAdd.js (hunk 4 above)
Comment (tzhenia) [reply src/main/resources/static/src/views/Petitions/PetitionAdd.js]: Thx, I removed it.

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/dto/PetitionDto.java (hunk 3 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/dto/PetitionDto.java]: Thank you! Done.

**Comment by tzhenia** [comment]
I think, pull requests is ready for marge

**Review by Michaluch — APPROVED** [review]

**Pull request merged**